spython/utils/misc.py

```diff
@@ -0,0 +1,31 @@
+import os
+
+def setEnvVar(name, value):
+    """Set or unset an environment variable"""
+    if value is None:
+        if name in os.environ:
+            del os.environ[name]
+    else:
+        os.environ[name] = value
+
+class ScopedEnvVar(object):
+    """Temporarly change an environment variable
+
+    Usage:
+        with ScopedEnvVar("FOO", "bar"):
+            print(os.environ["FOO"]) # "bar"
+        print(os.environ["FOO"]) # <oldvalue>
+    """
+    
+    def __init__(self, name, value):
+        self.name = name
+        self.value = value
+        self.oldValue = None
+    
+    def __enter__(self):
+        self.oldValue = os.environ.get(self.name)
```

Comment:
I'd appreciate a quick note here on enter and exit.

Reply:
This methods are not meant to be called but internal details of the class. The class is described in the docstring and the 3 code lines of this function IMO explain the working better than comments. So I'd rather not add docstrings here